core/src/engine.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "knoting/engine.h"
+
+namespace knot {
+
+Engine::Engine() {
+    m_windowModule = std::make_shared<knot::Window>(m_windowWidth, m_windowHeight, m_windowTitle);
```

Comment:
Do we want to create the main window in here? I understand that we want to add the systems in here, but should we decouple the main window creation to another place?
Or is this temporary?

Reply:
I was was thinking that the window should be a submodule as the pollevents / other callbacks / recreate swapchain are managed somewhat often, but we can fully decouple this

Reply:
Would this involve creating a constructor which takes in a reference to the created window?
Does this mean we would then be creating this window in main or abstracting to another class?
Is that even necessary?

Reply:
I would argue it is necessary if we want to support in the future multiple window creation. But it can perfectly be created here, exposing a way to interact with the windows.